pixeltable/functions/video.py

```diff
@@ -221,66 +223,471 @@ def get_metadata(video: pxt.Video) -> dict:
+            segment_path = f'{base_path}_segment_{len(result)}.mp4'
+            cmd = av_utils.ffmpeg_clip_cmd(str(video), segment_path, start_time, duration)
+
+            _ = subprocess.run(cmd, capture_output=True, text=True, check=True)
+            segment_duration = av_utils.get_video_duration(segment_path)
+            if segment_duration == 0.0:
+                # we're done
+                pathlib.Path(segment_path).unlink()
+                return result
+            result.append(segment_path)
+            start_time += segment_duration  # use the actual segment duration here, it won't match duration exactly
+
+        return result
+
+    except subprocess.CalledProcessError as e:
+        # clean up partial results
+        for segment_path in result:
+            pathlib.Path(segment_path).unlink()
+        _handle_ffmpeg_error(e)
+
+
+@pxt.udf(is_method=True)
+def concat_videos(videos: list[pxt.Video]) -> pxt.Video:
+    """
+    Merge multiple videos into a single video.
+
+    __Requirements:__
+
+    - `ffmpeg` needs to be installed and in PATH
+
+    Args:
+        videos: List of videos to merge.
+
+    Returns:
+        A new video containing the merged videos.
+    """
+    if len(videos) == 0:
+        raise pxt.Error('concat_videos(): empty argument list')
+    if not shutil.which('ffmpeg'):
+        raise pxt.Error('ffmpeg is not installed or not in PATH. Please install ffmpeg to use concat_videos().')
+
+    # Check that all videos have the same resolution
+    resolutions: list[tuple[int, int]] = []
+    for video in videos:
+        metadata = av_utils.get_metadata(str(video))
+        video_stream = next((stream for stream in metadata['streams'] if stream['type'] == 'video'), None)
+        if video_stream is None:
+            raise pxt.Error(f'concat_videos(): file {video} has no video stream')
+        resolutions.append((video_stream['width'], video_stream['height']))
+
+    # check for divergence
+    first_res = resolutions[0]
+    for i, res in enumerate(resolutions[1:], start=1):
+        if res != first_res:
+            raise pxt.Error(
+                f'concat_videos(): requires that all videos have the same resolution, but '
+                f'video 0 ({videos[0]}) has resolution {first_res[0]}x{first_res[1]} and '
+                f'video {i} ({videos[i]}) has resolution {res[0]}x{res[1]}.'
+            )
+
+    # ffmpeg -f concat needs an input file list
+    filelist_path = TempStore.create_path(extension='.txt')
+    with filelist_path.open('w', encoding='utf-8') as f:
+        for video in videos:
+            f.write(f"file '{video}'\n")
+
+    output_path = TempStore.create_path(extension='.mp4')
+
+    try:
+        # First attempt: fast copy without re-encoding (works for compatible formats)
+        cmd = ['ffmpeg', '-f', 'concat', '-safe', '0', '-i', str(filelist_path), '-c', 'copy', '-y', str(output_path)]
+        _logger.debug(f'concat_videos(): {" ".join(cmd)}')
+        try:
```

Comment:
As an alternative to `try/except` here, one could call `run()` with `check=False` and check the result code after it completes.